### PR TITLE
glcontext_*: error out early if OpenGLES is not supported

### PR DESCRIPTION
**Pull request by mbouron-gpsw**

### CODE REVIEW

File: libnodegl/glcontext_wgl.c [hunk 1]
```diff
@@ -49,6 +49,11 @@ static int wgl_init(struct glcontext *ctx, uintptr_t display, uintptr_t window,
 {
     struct wgl_priv *wgl = ctx->priv_data;
 
+    if (ctx->backend != NGL_BACKEND_OPENGL) {
+        LOG(ERROR, "unsupported backend: %s, only OpenGL is supported by WGL");
```
Comment (cboesch-gpsw) [inline libnodegl/glcontext_wgl.c:53]: Our implem, or WGL itself?

File: libnodegl/glcontext_nsgl.m [hunk 2]
```diff
@@ -42,6 +42,11 @@ static int nsgl_init(struct glcontext *ctx, uintptr_t display, uintptr_t window,
 {
     struct nsgl_priv *nsgl = ctx->priv_data;
 
+    if (ctx->backend != NGL_BACKEND_OPENGL) {
+        LOG(ERROR, "unsupported backend: %d, only OpenGL is supported by NSGL", ctx->backend);
```
Comment (cboesch-gpsw) [inline libnodegl/glcontext_nsgl.m:46]: Same question.

File: libnodegl/glcontext.c [hunk 3]
```diff
@@ -113,8 +113,8 @@ static int glcontext_load_functions(struct glcontext *glcontext)
 
 static int glcontext_probe_version(struct glcontext *glcontext)
 {
-    GLint major_version;
-    GLint minor_version;
+    GLint major_version = 0;
+    GLint minor_version = 0;
```
Comment (cboesch-gpsw) [inline libnodegl/glcontext.c:117]: Fine with me but note that with `DEBUG_GL=yes` we get a crash

File: libnodegl/glcontext_wgl.c (hunk 1 above)
Comment (mbouron-gpsw) [reply libnodegl/glcontext_wgl.c:53]: In this case, it is our implementation that do not support GLES.

Comment (mbouron-gpsw) [reply libnodegl/glcontext_wgl.c:53]: I'll try enable its support in a separate PR.

Comment (mbouron-gpsw) [reply libnodegl/glcontext_wgl.c:53]: Would renaming the error message to "unsupported backend: %s, only the OpenGL backend is supported" be ok with you ? (Same for NSGL).

Comment (cboesch-gpsw) [reply libnodegl/glcontext_wgl.c:53]: Yes, the wording would be more correct; you could even add "currently" before "supported".

File: libnodegl/glcontext_nsgl.m (hunk 2 above)
Comment (mbouron-gpsw) [reply libnodegl/glcontext_nsgl.m:46]: NSGL only supports OpenGL.

Comment (cboesch-gpsw) [reply libnodegl/glcontext_nsgl.m:46]: Then this message is correct.

File: libnodegl/glcontext.c (hunk 3 above)
Comment (mbouron-gpsw) [reply libnodegl/glcontext.c:117]: I will fix that issue in a separate PR (as well as adding GLES support for WGL).

File: libnodegl/glcontext_wgl.c (hunk 1 above)
Comment (mbouron-gpsw) [reply libnodegl/glcontext_wgl.c]: Done.

File: libnodegl/glcontext_wgl.c [hunk 4]
```diff
@@ -49,6 +49,11 @@ static int wgl_init(struct glcontext *ctx, uintptr_t display, uintptr_t window,
 {
     struct wgl_priv *wgl = ctx->priv_data;
 
+    if (ctx->backend != NGL_BACKEND_OPENGL) {
+        LOG(ERROR, "unsupported backend: %s, only OpenGL is currently supported");
```
Comment (cboesch-gpsw) [inline libnodegl/glcontext_wgl.c:53]: This is missing an argument.

File: libnodegl/glcontext_nsgl.m [hunk 5]
```diff
@@ -42,6 +42,11 @@ static int nsgl_init(struct glcontext *ctx, uintptr_t display, uintptr_t window,
 {
     struct nsgl_priv *nsgl = ctx->priv_data;
 
+    if (ctx->backend != NGL_BACKEND_OPENGL) {
+        LOG(ERROR, "unsupported backend: %d, only OpenGL is supported", ctx->backend);
```
Comment (cboesch-gpsw) [inline libnodegl/glcontext_nsgl.m:46]: Better with the original here with the mention of "NSGL" to confirm it's not a limitation in node.gl

File: libnodegl/glcontext_wgl.c (hunk 4 above)
Comment (mbouron-gpsw) [reply libnodegl/glcontext_wgl.c]: Fixed.

File: libnodegl/glcontext_nsgl.m (hunk 5 above)
Comment (mbouron-gpsw) [reply libnodegl/glcontext_nsgl.m]: Done.

**Review by cboesch-gpsw — APPROVED** [review]

**Pull request merged**